### PR TITLE
Eledger 109 get transaction using Ledger Id and Borrower Id

### LINKED ISSUE #16: Eledger-110: Create Api to get list of Transactions for landerId and borrowerID
create Api which fetch all the list of Transactions for a particular walletID

### PR DESCRIPTION
**Pull request by SahilKharya**

### CODE REVIEW

File: src/main/java/in/pune/royforge/eledgerapi/controller/TransactionController.java [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
+@RequestMapping("/tansaction")
 public class TransactionController {
-		
+
 	@Autowired
-	WalletService walletEntityService;
-	
+	TransactionService transactionService;
+
+	@RequestMapping(value = "/{lenderId}/{borrowerId}", method = RequestMethod.GET)
```
Comment (abkesarwani) [inline src/main/java/in/pune/royforge/eledgerapi/controller/TransactionController.java]: update RequestMapping value to more meaning full

File: src/main/java/in/pune/royforge/eledgerapi/controller/TransactionController.java [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
+	public List<Transaction> walletTransactionsLog(@PathVariable(value = "lenderId") String lenderId,
```
Comment (abkesarwani) [inline src/main/java/in/pune/royforge/eledgerapi/controller/TransactionController.java]: change method name to more meaning full name

Comment (SahilKharya) [reply src/main/java/in/pune/royforge/eledgerapi/controller/TransactionController.java]: done

File: src/main/java/in/pune/royforge/eledgerapi/data/repo/ITransactionLogRepository.java [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.PagingAndSortingRepository;
 
 import in.pune.royforge.eledgerapi.data.entity.TransactionEntity;
 
-public interface ITransactionLogRepository extends PagingAndSortingRepository<TransactionEntity, String> {
+public interface ITransactionLogRepository extends PagingAndSortingRepository<TransactionEntity, Long> {
+
+	@Query("FROM TransactionEntity WHERE lenderId = ?1 AND borrowerId = ?2")
+	Iterable<TransactionEntity> transactionsList(String lenderId, String borrowerId);
```
Comment (abkesarwani) [inline src/main/java/in/pune/royforge/eledgerapi/data/repo/ITransactionLogRepository.java]: change return type of transactionsList method-->

List<TransactionEntity> transactionsList(String lenderId, String borrowerId);

File: src/main/java/in/pune/royforge/eledgerapi/data/dao/TransactionDAOImpl.java [hunk 4]
```diff
@@ -12,11 +12,30 @@
...
+	/*
+	 * Method to return a list List contains the details of transaction done between
+	 * a particular lender ID and borrower ID Lender Id and Borrower ID is passed as
+	 * argument to this method
+	 * 
+	 */
+	@Override
+	public List<Transaction> getTransactionsUsingLenderIdAndBorrowerId(String lenderId, String borrowerId) {
+		List<Transaction> transactions = new ArrayList<>();
+		Iterable<TransactionEntity> transactionsList = transactionLogRepository.transactionsList(lenderId, borrowerId);
```
Comment (abkesarwani) [inline src/main/java/in/pune/royforge/eledgerapi/data/dao/TransactionDAOImpl.java]: check isEmpty now like -----> 

List<Transaction> transactions = new ArrayList<>();
		List<TransactionEntity> transactionEntities = transactionLogRepository.transactionsList(lenderId, borrowerId);
		if (transactionEntities.isEmpty()) {
			for (TransactionEntity transaction1 : transactionEntities) {
				Transaction transactionInfo = new Transaction();
				setTransactionData(transaction1, transactionInfo);
				transactions.add(transactionInfo);
			}
		}
		return transactions;

File: src/main/java/in/pune/royforge/eledgerapi/data/dao/TransactionDAOImpl.java [hunk 5]
```diff
@@ -12,11 +12,30 @@
...
+		for (TransactionEntity transaction1 : transactionsList) {
```
Comment (abkesarwani) [inline src/main/java/in/pune/royforge/eledgerapi/data/dao/TransactionDAOImpl.java]: change variable name transaction1 ---> transaction

File: src/main/java/in/pune/royforge/eledgerapi/data/entity/TransactionEntity.java [hunk 6]
```diff
@@ -103,8 +102,8 @@ public String getTxnType() {
 		return txnType;
 	}
 
-	public void setTxnType(String txnType) {
-		this.txnType = txnType;
+	public void setTxnType(String string) {
```
Comment (abkesarwani) [inline src/main/java/in/pune/royforge/eledgerapi/data/entity/TransactionEntity.java:105]: change variable name "string"

**Review by abkesarwani — APPROVED** [review]

**Pull request merged**